src/server/project.ts

```diff
@@ -2234,6 +2234,11 @@ namespace ts.server {
                 const basePath = this.getCurrentDirectory();
 
                 const getPnpPath = (path: string) => {
+                    // When the path is absolute, pnp might resolve it wrongly for any
+                    // file that is present within the `workspace:.` folder.
+                    // So defaulting to using `path` when it is absolute.
+                    if (pathIsAbsolute(path)) return path;

```

Comment:
Should this be `ts.pathIsAbsolute`?

I'm not sure if `pathIsAbsolute` exists in the global scope here. I needed to add it as `ts.pathIsAbsolute` in the generated `.js` file, otherwise the language service would crash.

Reply:
The compiled JS should have `ts.pathIsAbsolute` because it is declared in the `ts` namespace and available to any module/function/class defined within the same namespace.